projects/power_distribution/src/main.c

```diff
@@ -1,6 +1,14 @@
 // Uncomment this line to force firmware to run as front or rear power distribution.
 // #define FORCE_IS_FRONT_POWER_DISTRIBUTION true
 
+// SMOKE TEST MODES: each macro is a different smoke test you can run.
+
+#define SMOKE_TEST_MODE_NORMAL 0               // run as normal PD
+#define SMOKE_TEST_MODE_CURRENT_MEASUREMENT 1  // see smoke_current_measurement.c
```

Comment:
So if I'm reading correctly, the idea that as we expand upon smoke testing infrastructure for PD that the user can adjust the "mode" by changing the defined macro here. 

Since we are looking at developing these smoke tests for each project, I am kind of thinking that this PR will represent the de-facto for how we go about it, and as new ones are written they will follow how this is setup, so I want to discuss a few things.

I like the idea of being able to change smoke tests by just adjusting the macro, but I am worried about cluttering the program as we increase to more than one smoke test.  If we are thinking multiple smoke tests per project, the same way that it is set up here, then it might make sense to have all smoke tests of the same function type (no return, no params), and have a lookup table defined in a `smoke_<project>.h` header file along with the macros defining each mode.

Then all we need in main is the one line which sets `SMOKE_TEST_MODE` and the one check against MODE_NORMAL in main, otherwise call whatever is in the lookup table. (effectively the same as what is happening now, just scales with more smoke tests)

This is more a factor in PD and other big projects, as increasing complexity just makes them harder to read and debug.

Reply:
This makes a lot of sense, I'll change to this. I am a bit worried that `smoke_<project>.h` might look like another smoke test instead of a smoke test control header, so I think the name `smoketests_<project>.h` might fit better.